### PR TITLE
Media Foundation S/W Decoder Support for Windows Platform [MAGMA-3052]

### PR DESCRIPTION
**Pull request by rjohnson-gpsw**
Added the following implementation and decoder declarations from those originally authored by WM4/Pawel Wegner (c. 2017)

    AAC, H264, HEVC.
The remaining original codecs where excluded as not needed:
    AC3, EAC3, MJPEG, MP1, MP2, MP3, MPEG2, MPEG4, MSMPEG4v1, MSMPEG4v2, MSMPEG4v3,
    VC1, WMAV1, WMAV2, WMALossless, WMAPro, WMAVoice, WMV1, WMV2, WMV3

See (mfdec.c and hwcontext_mediafoundation.c)

Added configure script changes to include these decoders and dependencies for MF libs.

Moved mf_create and mf_load_library out of mfenc.c and into mf_utils.c to create a common implementation for both decoder and encoder use.

Included the original MF hwaccel implementation, pixel description and format declarations

[MAGMA-3052](https://goproinc.atlassian.net/browse/MAGMA-3052)

[MAGMA-3052]: https://goproinc.atlassian.net/browse/MAGMA-3052?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libavformat/demux.c [hunk 1]
```diff
@@ -70,12 +70,34 @@ static const AVCodec *find_probe_decoder(AVFormatContext *s, const AVStream *st,
...
+    /*  force MF variants for decoding on Windows */
+    #if CONFIG_H264_MF_DECODER
+    if (codec_id == AV_CODEC_ID_H264)
+        return avcodec_find_decoder_by_name("h264_mf");
+    #endif
+    #if CONFIG_HEVC_MF_DECODER
+    if (codec_id == AV_CODEC_ID_HEVC)
+        return avcodec_find_decoder_by_name("hevc_mf");
+    #endif
+    #if defined(_WIN32) && CONFIG_AAC_MF_DECODER
```
Comment (jstys-gpsw) [inline libavformat/demux.c]: Could drop the WIN32 check here, it's checked above

File: libavcodec/mfenc.c [hunk 2]
```diff
@@ -1209,10 +1110,9 @@ static int mf_close(AVCodecContext *avctx)
 static int mf_init(AVCodecContext *avctx)
 {
     int ret;
-    if ((ret = mf_load_library(avctx)) == 0) {
-           if ((ret = mf_init_encoder(avctx)) == 0) {
-                return 0;
-        }
+    MFContext *c = avctx->priv_data;
```
Comment (jstys-gpsw) [inline libavcodec/mfenc.c]: variable `c` is unused here

**Review by jstys-gpsw — COMMENTED** [review]
Minor comments from me so far

File: libavcodec/mfenc.c (hunk 2 above)
Comment (rjohnson-gpsw) [reply libavcodec/mfenc.c]: Thanks, I think this was a byproduct of my refactor!

File: libavformat/demux.c (hunk 1 above)
Comment (rjohnson-gpsw) [reply libavformat/demux.c]: Thanks, leftover from refactor.

**Review by rlavarec-gpsw — APPROVED** [review]

**Review by wmacdonald-gpsw — APPROVED** [review]

**Pull request closed without merging**